ckan/logic/action/create.py

```diff
@@ -300,8 +300,11 @@ def resource_create(context, data_dict):
         _get_action('package_update')(context, pkg_dict)
         context.pop('defer_commit')
     except ValidationError, e:
-        errors = e.error_dict['resources'][-1]
-        raise ValidationError(errors)
+        if ('resources' in e.error_dict):
+            errors = e.error_dict['resources'][n]
+            raise ValidationError(errors)
```

Comment:
how about
```python
try:
    raise ValidationError(e.error_dict['resources'][-1])
except (KeyError, IndexError):
    raise ValidationError(e.error_dict)
```

that will catch the case of an error in a different resource than the one submitted too